### PR TITLE
Move contributing info to CONTRIBUTING

### PR DESCRIPTION
**Pull request by Manishearth**
The README is long and doesn't immediately indicate that there is a procedure to follow for pull requests. Moving it out to CONTRIBUTING (where most projects put "how to contribute" information) makes it easier to find. Github also puts up a banner when making pull requests when there is a contributing file that hasn't been read.

I also updated the coding style link and added the blurb about "pull requests are okay however please email rt@" which was missing from the section.

### CODE REVIEW

File: CONTRIBUTING [hunk 1]
```diff
@@ -0,0 +1,43 @@
+HOW TO CONTRIBUTE TO OpenSSL
+----------------------------
+
+Development is coordinated on the openssl-dev mailing list (see
+http://www.openssl.org for information on subscribing). If you
+would like to submit a patch, send it to openssl-bugs@openssl.org with
```
Comment (mattcaswell) [inline CONTRIBUTING]: I'd make that email address "rt@openssl.org" for consistency

Comment (Manishearth) [inline CONTRIBUTING]: Fixed!

File: CONTRIBUTING [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+----------------------------
+
+Development is coordinated on the openssl-dev mailing list (see
+http://www.openssl.org for information on subscribing). If you
+would like to submit a patch, send it to rt@openssl.org with
+the string "[PATCH]" in the subject. Please be sure to include a
+textual explanation of what your patch does.
+
+You can also make GitHub pull requests. If you do this, please also send
+mail to rt@openssl.org with a link to the PR so that we can more easily
```
Comment (mattcaswell) [inline CONTRIBUTING]: Lets make that "with a brief description and a link"

Comment (Manishearth) [inline CONTRIBUTING]: (fixed)

Should I myself send an updated description for the other PR I sent to the list, or is it fine for now?

**Comment by mattcaswell** [comment]
+1

File: README [hunk 3]
```diff
@@ -158,40 +158,4 @@
...
- Our preferred format for changes is "diff -u" output. You might
- generate it like this:
-
- # cd openssl-work
- # [your changes]
- # ./Configure dist; make clean
- # cd ..
- # diff -ur openssl-orig openssl-work > mydiffs.patch
-
+ See https://github.com/openssl/openssl/blob/master/CONTRIBUTING
```
Comment (levitte) [inline README]: Hmmm, I understand your though, but I think that it should either be

    See CONTRIBUTING

or we should move CONTRIBUTING somewhere completely different, such as wiki.openssl.org.

For now, I would prefer the former.

Comment (Manishearth) [inline README]: Fixed

**Comment by levitte** [comment]
+1
merged and pushed

**Pull request closed without merging**